mycroft/configuration/config.py

```diff
@@ -188,163 +197,272 @@ def _log_old_location_deprecation(old_user_config=OLD_USER_CONFIG):
     LOG.warning(" Please move it to " + get_xdg_config_save_path())
 
 
-def _get_system_constraints():
-    # constraints must come from SYSTEM config
-    # if not defined then load the DEFAULT constraints
-    # these settings can not be set anywhere else!
-    return LocalConf(SYSTEM_CONFIG).get("system") or \
-           LocalConf(DEFAULT_CONFIG).get("system") or \
-           {}
+class CoreConfig(dict):
+    """Namespace for operations on the configuration singleton."""
+    __patch = LocalConf(None)  # Patch config that skills can update to override config
+    bus = None
+    default = LocalConf(DEFAULT_CONFIG)
+    system = LocalConf(SYSTEM_CONFIG)
+    remote = RemoteConf()
+    # This includes both the user config and
+    # /etc/xdg/mycroft/mycroft.conf
+    xdg_configs = [LocalConf(p) for p in get_xdg_config_locations()]
+    _old_user = LocalConf(OLD_USER_CONFIG)
+
+    def __setitem__(self, key, value):
+        CoreConfig.__patch[key] = value
+        # sync with other processes connected to bus
+        if CoreConfig.bus:
+            CoreConfig.bus.emit(Message("configuration.patch",
+                                        {"config": {key: value}}))
+
+    def __getitem__(self, item):
+        cfg = CoreConfig.load_config_stack()
+        return cfg.get(item)
+
+    def __str__(self):
+        cfg = CoreConfig.load_config_stack()
+        try:
+            return json.dumps(cfg, sort_keys=True)
+        except:
+            return str(cfg)
 
+    def __dict__(self):
+        return CoreConfig.load_config_stack()
 
-class Configuration:
```

Comment:
Just skimming this PR, but would it be possible for a backwards-compat `Configuration` class to remain here with a `get` method? Mostly thinking about skills and plugins that may have referenced this directly and also considering how this might help troubleshoot unit test failures if the legacy methods are available